webserver/webserver/src/main/java/io/helidon/webserver/RequestPredicate.java

```diff
@@ -16,226 +16,508 @@
+            return headerValue.isPresent()
+                    && headerValue.get().equals(value);
+        });
+    }
 
     /**
-     * Accepts only requests with specified cookie containing valid value.
+     * Accept requests only when the specified query parameter exists.
      *
-     * @param cookieName Header name.
-     * @param value Predicate for a cookie value.
-     * @return New enhanced instance.
+     * @param name query parameter name
+     * @return composed predicate representing the logical expression of the previous
+     * predicate expression <b>and</b> the logical expression implemented by this
+     * method
+     * @throws NullPointerException if the specified name is null
      */
-    RequestPredicate containsCookie(String cookieName, String value);
+    public RequestPredicate containsQueryParameter(final String name) {
+        Objects.requireNonNull(name, "query param name");
+        return and((req) -> req.queryParams()
+                .first(name)
+                .isPresent());
+    }
 
     /**
-     * Accepts only requests accepting any of specified content types.
+     * Accept requests only when the specified query parameter is valid.
      *
-     * @param contentType Content type.
-     * @return New enhanced instance.
+     * @param name query parameter name
+     * @param predicate to match the query parameter value
+     * @return composed predicate representing the logical expression of the previous
+     * predicate expression <b>and</b> the logical expression implemented by this
+     * method
+     * @throws NullPointerException if the specified name or predicate is null
      */
-    RequestPredicate accepts(String... contentType);
+    public RequestPredicate containsQueryParameter(final String name,
+            final Predicate<String> predicate) {
+
+        Objects.requireNonNull(name, "query param name");
+        Objects.requireNonNull(predicate, "query param predicate");
+        return and((req) -> req.queryParams()
+                .all(name)
+                .stream()
+                .anyMatch(predicate));
+    }
 
     /**
-     * Accepts only requests accepting any of specified content types.
+     * Accept requests only when the specified query parameter contains a given
+     * value.
      *
-     * @param contentType Content type.
-     * @return New enhanced instance.
+     * @param name query parameter name
+     * @param value expected query parameter value
+     * @return composed predicate representing the logical expression of the previous
+     * predicate expression <b>and</b> the logical expression implemented by this
+     * method
+     * @throws NullPointerException if the specified name or value is null
      */
-    RequestPredicate accepts(MediaType... contentType);
+    public RequestPredicate containsQueryParameter(final String name,
+            final String value) {
+
+        Objects.requireNonNull(name, "query param name");
```

Comment:
Could this method simply delegate to the other variant?

Reply:
Yes it can, I'll make the change